Fix string truncation in belle-sip dns.c

The hex string "0123456789abcdef" is 17 bytes including NUL terminator,
but the array is only 16 bytes. Increase array size to 17.

Upstream-Status: Inappropriate [workaround for GCC 15 compatibility]

--- a/belle-sip/src/dns/dns.c
+++ b/belle-sip/src/dns/dns.c
@@ -2790,7 +2790,7 @@ size_t dns_aaaa_arpa(void *_dst, size_t lim, const struct dns_aaaa *aaaa) {
 } /* dns_aaaa_cmp() */
 
 size_t dns_aaaa_arpa(void *_dst, size_t lim, const struct dns_aaaa *aaaa) {
-	static const unsigned char hex[16] = "0123456789abcdef";
+	static const unsigned char hex[17] = "0123456789abcdef";
 	struct dns_buf dst = DNS_B_INTO(_dst, lim);
 	unsigned nyble;
 	int i, j;
@@ -3382,7 +3382,7 @@ size_t dns_sshfp_print(void *_dst, size_t lim, struct dns_sshfp *fp) {
 } /* dns_sshfp_cmp() */
 
 size_t dns_sshfp_print(void *_dst, size_t lim, struct dns_sshfp *fp) {
-	static const unsigned char hex[16] = "0123456789abcdef";
+	static const unsigned char hex[17] = "0123456789abcdef";
 	struct dns_buf dst = DNS_B_INTO(_dst, lim);
 	size_t i;
 
